src/pudl/transform/ferc1.py

```diff
@@ -1021,7 +1297,498 @@ def process_xbrl_metadata(self, xbrl_metadata_json) -> pd.DataFrame:
+                    "source_table"
+                ),
+            )
         )
+        xbrl_factoid_name_map = {
+            xbrl_factoid_name_og: self.rename_column_xbrl_to_pudl(xbrl_factoid_name_og)
+            for xbrl_factoid_name_og in tbl_meta.xbrl_factoid
+        }
+        tbl_meta.xbrl_factoid = tbl_meta.xbrl_factoid.map(xbrl_factoid_name_map)
+
+        def rename_calculation_components(calc: str) -> str:
+            # apply the rename for the "name" element of all of the calc components
+            renamed_calc = [
+                {
+                    k: self.rename_column_xbrl_to_pudl(v) if k == "name" else v
+                    for (k, v) in calc_component.items()
+                }
+                for calc_component in json.loads(calc)
+            ]
+            return json.dumps(renamed_calc)
+
+        tbl_meta.calculations = tbl_meta.calculations.apply(
+            rename_calculation_components
+        )
+        tbl_meta = (
+            self.apply_xbrl_calc_fixes(tbl_meta)
+            .pipe(self.remove_duplicated_calc_components)
+            .pipe(self.add_calc_correction_components)
+        )
+        tbl_meta.calculations = tbl_meta.calculations.astype(pd.StringDtype())
+
+        return tbl_meta
+
+    def rename_column_xbrl_to_pudl(
+        self,
+        col_name_xbrl: str,
+    ) -> str:
+        """Rename a column name from orignal XBRL name to the transformed PUDL name.
+
+        There are several transform params that either explicitly or implicity rename
+        columns:
+        * :class:`RenameColumnsFerc1`
+        * :class:`WideToTidySourceFerc1`
+        * :class:`UnstackBalancesToReportYearInstantXbrl`
+        * :class:`ConvertUnits`
+
+        This method attempts to use the table params to translate a column name.
+
+        Note: Instead of doing this for each individual column name, we could compile a
+        rename dict for the whole table with a similar processand then apply it for each
+        group of columns instead of running through this full process every time. If
+        this took longer than...  ~5 ms on a single table w/ lots of calcs this would
+        probably be worth it for simplicity.
+        """
+        col_name_new = col_name_xbrl
+        for rename_stage in self.params.rename_dicts_xbrl:
+            col_name_new = str(rename_stage.columns.get(col_name_new, col_name_new))
+
+        for value_type in self.params.wide_to_tidy_value_types:
+            if col_name_new.endswith(f"_{value_type}"):
+                col_name_new = re.sub(f"_{value_type}$", "", col_name_new)
+
+        if self.params.unstack_balances_to_report_year_instant_xbrl:
+            # TODO: do something...? add starting_balance & ending_balance suffixes?
+            pass
+        if self.params.convert_units:
+            # TODO: use from_unit -> to_unit map. but none of the $$ tables have this rn.
+            pass
```

Comment:
like yes in a perfect world but no because we are now applying this rename across the board regardless. these types of renaming/metadata treatment was hopefully already being done within the table-specific `process_xbrl_metadata`.

Reply:
Are you saying that using `NotImplementedError` would break things here? I'm just suggesting that we guard against this method being used in an inappropriate context -- i.e. on tables where we haven't actually defined all of the necessary renames -- which I had the impression wouldn't include the tables we're using it on right now (i.e. it would never get into these "if" clauses)

Reply:
this is being applied to all tables bc it is being applied within the default version of `process_xbrl_metadata`. the tables with these two treatments are the plant tables. I think the way to add a warning about this is if the table is getting `merge_xbrl_metadata` & either of these two treatments are present.

bc that is  the condition where this rename not being applied would actually matter.